src/main/java/uk/gov/justice/hmpps/prison/repository/mapping/DataClassByColumnRowMapper.kt

```diff
@@ -69,4 +76,11 @@ class DataClassByColumnRowMapper<T>(mappedClass: Class<T>) : BeanPropertyRowMapp
     }
     return BeanUtils.instantiateClass(mappedConstructor, *args)
   }
+
+  override fun getColumnValue(rs: ResultSet, index: Int, paramType: Class<*>): Any? {
+    return when (paramType) {
+      Boolean::class.javaObjectType -> "Y" == rs.getObject(index) || rs.getBoolean(index)
```

Comment:
note the `.javaObjectType` which translates to `java.lang.Boolean` so we have the same functionality as in the [StandardBean](https://github.com/ministryofjustice/prison-api/blob/1e1ecef6c358db4fd5af4b6d456484f9e725716f/src/main/java/uk/gov/justice/hmpps/prison/repository/mapping/StandardBeanPropertyRowMapper.java#L71) version.  Didn't need the local date (time) versions as spring translates these fine these days